### PR TITLE
RUBY-2846 Add _bson_to_i method to ActiveSupport::TimeWithZone class

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

**Comment by p-mongo** [comment]
This should not be necessary since https://github.com/jruby/jruby/issues/6104 is merged into JRuby 9.3 and Mongoid 8.0+ requires 9.3.

**Comment by Neilshweky** [comment]
> This should not be necessary since https://github.com/jruby/jruby/issues/6104 is merged into JRuby 9.3 and Mongoid 8.0+ requires 9.3.

There is definitely work to be done in Mongoid wrt this though, right? Removing _bson_to_i and just calling to_i if we're 9.3+?

File: lib/bson/time_with_zone.rb [hunk 1]
```diff
@@ -46,6 +46,16 @@ def to_bson(buffer = ByteBuffer.new)
     def bson_type
       ::Time::BSON_TYPE
     end
+
+    def _bson_to_i
```
Comment (p-mongo) [inline lib/bson/time_with_zone.rb:50]: Can you add `@api private` here please.

**Review by p-mongo — APPROVED** [review]
I changed my mind, since we don't detect jruby versions let's support correct behavior on 9.2.

When either bson-ruby is impossible to use on 9.2 or it's very clear to users this shouldn't be done, all _bson_to_i code should be removed.

File: lib/bson/time_with_zone.rb (hunk 1 above)
Comment (Neilshweky) [inline lib/bson/time_with_zone.rb:50]:
```suggestion
    # @api private
    def _bson_to_i
```

Comment (Neilshweky) [reply lib/bson/time_with_zone.rb:51]: done

**Pull request merged**